File: OrderTracker/frontend/order-tracker-ui/src/types/order.ts
export interface OrderNote {
  author: string;
  message: string;
  createdAt: string;
}

export interface OrderTimeline {
  status: string;
  timestamp: string;
}

export interface Order {
  id: string;
  email: string;
  mobile: string;
  status: string;
  notes: OrderNote[];
  timeline: OrderTimeline[];
}

File: OrderTracker/src/OrderTracker.Core/Entities/Order.cs
using OrderTracker.Core.Enums;

namespace OrderTracker.Core.Entities;

public class Order
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Email { get; set; } = string.Empty;
    public string Mobile { get; set; } = string.Empty;

    public OrderStatus Status { get; set; } = OrderStatus.PLACED;

    public List<OrderNote> Notes { get; set; } = new();
    public List<OrderTimeline> Timeline { get; set; } = new();
}